operator/operator.go

```diff
@@ -457,6 +479,17 @@ func (o *Operator) rescaleStatefulSet(ctx context.Context, sts *appsv1.StatefulS
 				return fmt.Errorf("failed to drain pod %s/%s: %v", pod.Namespace, pod.Name, err)
 			}
 			log.Infof("Pod %s/%s drained", pod.Namespace, pod.Name)
+			// check if we need to opt-out of the loop.
+			newSR, err := srg.Get(ctx)
+			if err != nil {
+				return fmt.Errorf("failed to refresh EDS: %v", err)
+			}
+			newDesiredReplicas := int(newSR.Replicas())
+			if newDesiredReplicas > desiredReplicas {
+				log.Infof("EDS %s/%s target scaling definition changed from %d to %d, aborting scale-down", sr.Namespace(), sr.Name(), desiredReplicas, newDesiredReplicas)
+				return nil
+			}
+
```

Comment:
nit: I would propose to move this to the beginning of the loop just to make it more clear that we check this before scaling down anything.

Reply:
done.